### PR TITLE
Feature/dst 99 remove fields event

### PR DESCRIPTION
**Pull request by tomtollafield**

### CODE REVIEW

File: DFC.ServiceTaxonomy.GraphSync/Activities/RemoveFieldFromContentItemsTask.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+        {
+            string? contentTypeToSync, fieldToRemove;
+            ValidateParameters(workflowContext, out contentTypeToSync, out fieldToRemove);
+
+            _contentDefinitionService.RemoveFieldFromPart(fieldToRemove, contentTypeToSync);
+
+            var b = _contentManager.GetPartDefinition(contentTypeToSync);
```
Comment (lazcool) [inline DFC.ServiceTaxonomy.GraphSync/Activities/RemoveFieldFromContentItemsTask.cs]: Would you mind explaining this check for me?
Also, could we have a more descriptive variable name and is `InvalidCastException` an appropriate exception to throw?

File: DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Parts/EponymousPartGraphSyncer.cs [hunk 2]
```diff
@@ -78,9 +78,20 @@ public EponymousPartGraphSyncer(IContentManager contentManager)
...
+
+                string? keyName = ((JProperty)field).Name;
+
                 if (firstProperty == null)
                     continue;
 
+                if (!contentTypePartDefinition.PartDefinition.Fields.Any(z => z.Name.Equals(keyName, StringComparison.OrdinalIgnoreCase)))
+                {
+#pragma warning disable CS8625 // Cannot convert null literal to non-nullable reference type.
+                    mergeNodeCommand.Properties.Add(await _graphSyncHelper!.PropertyName(keyName), null);
```
Comment (lazcool) [inline DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Parts/EponymousPartGraphSyncer.cs]: Is this really necessary?
I think properties with a null value aren't actually created on the node in neo.
If there is a good reason, we'll have to make sure the change doesn't go missing during merge, as the epoonymous graph syncer has been refactored in an outstanding pr.

File: DFC.ServiceTaxonomy.GraphSync/Activities/RemoveFieldFromContentItemsTask.cs (hunk 1 above)
Comment (tomtollafield) [reply DFC.ServiceTaxonomy.GraphSync/Activities/RemoveFieldFromContentItemsTask.cs]: Aha! This is POC code that needs removing, I was trying to see which point the cache wasn't being updated and missed removing this

File: DFC.ServiceTaxonomy.GraphSync/Managers/CustomContentDefinitionManager.cs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        /// </summary>
+        private ContentDefinitionRecord LoadContentDefinitionRecord() =>
+            _contentDefinitionStore.LoadContentDefinitionAsync().GetAwaiter().GetResult();
+
+        private ContentTypeDefinition Build(ContentTypeDefinitionRecord source, IList<ContentPartDefinitionRecord> partDefinitionRecords)
+        {
+            if (source == null)
+            {
+#pragma warning disable CS8603 // Possible null reference return.
+                return null;
```
Comment (lazcool) [inline DFC.ServiceTaxonomy.GraphSync/Managers/CustomContentDefinitionManager.cs]: I think the warning is valid here. The method returns a non-nullable reference type, but the method returns null.
As the method is private, we should be able to return `ContentTypeDefinition?`.

File: DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Parts/EponymousPartGraphSyncer.cs (hunk 2 above)
Comment (tomtollafield) [reply DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Parts/EponymousPartGraphSyncer.cs]: Now that the cache is clearing we may be able to simply continue here - I shall give it a go. The problem I had was that not passing the property as null to Neo, it was still appearing, but that could be due to caching.

File: DFC.ServiceTaxonomy.GraphSync/Services/CustomContentDefinitionService.cs [hunk 4]
```diff
@@ -131,7 +131,16 @@ public IEnumerable<EditTypeViewModel> LoadTypes()
 
         public void RemoveFieldFromPart(string fieldName, string partName)
         {
-            _orchardCoreContentDefinitionService.RemoveFieldFromPart(fieldName, partName);
+            var typeBeingUpdated = _contentDefinitionManager.GetTypeDefinition(partName);
+
+            if (typeBeingUpdated != null && typeBeingUpdated.Parts.Any(x => x.Name == "GraphSyncPart"))
```
Comment (lazcool) [inline DFC.ServiceTaxonomy.GraphSync/Services/CustomContentDefinitionService.cs]: Could you change this to `nameof(GraphSyncPart)` to make it more refactor friendly.

**Pull request merged**